resource-managers/mesos/src/main/scala/org/apache/spark/scheduler/cluster/mesos/MesosCoarseGrainedSchedulerBackend.scala

```diff
@@ -173,6 +176,17 @@ private[spark] class MesosCoarseGrainedSchedulerBackend(
   override def start() {
     super.start()
 
+    // fetch DTs early enough to avoid metastore connection issue
+    // proxy user in cluster mode
+    if (sc.conf.getOption("spark.mesos.cluster.mode.proxyUser").isDefined) {
```

Comment:
This is another issue the customer faced before the proxy user implementation. We follow the Yarn approach here: https://github.com/apache/spark/pull/17335/files